src/front/wgsl/lexer.rs

```diff
@@ -13,7 +13,8 @@ fn consume_any(input: &str, what: impl Fn(char) -> bool) -> (&str, &str) {
     input.split_at(pos)
 }
 
-fn consume_number(input: &str) -> (&str, &str) {
+fn consume_number(input: &str) -> (Token, &str) {
+    //Note: I wish this function was simpler and faster...
```

Comment:
Could you explain why it isn't fast? Simpler I get, but as far as I see it is fast?

Reply:
We are checking for all these special `-`, `.`, `e`, symbols as we go, and now we are also scanning the string we just processed for the `.` after the fact. And after all that, we'll be doing this again to extract the actual number from the sub-string.